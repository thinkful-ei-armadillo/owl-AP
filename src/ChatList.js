import React from 'react';
import ChatItem from './ChatItem';
import './ChatList.css';

export default function ChatList(props) {
  function getParticipantObject(id) {
    return props.participants.find(element => element.id === id);
  }
  const chat = props.chatEvents;

  let chatJSX = chat.map((element, index) => {
    debugger;
    const participant = getParticipantObject(element.participantId);

    return (
      <ChatItem
        key={index}
        user={participant}
        type={element.type}
        message={element.message}
        time={element.time}
        timestamp={element.timestamp}
      />
    );
  });

  return <ul>{chatJSX}</ul>;
}
