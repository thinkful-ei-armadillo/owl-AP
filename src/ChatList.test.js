import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatList from './ChatList';
import store from './store';

describe('ParticipantList component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    //render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<ChatList 
        chatEvents={store.chatEvents}
        participants={store.participants} 
        />, div);

    //clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ChatList 
        chatEvents={store.chatEvents}
        participants={store.participants} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
