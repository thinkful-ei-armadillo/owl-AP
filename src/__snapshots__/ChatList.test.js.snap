// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantList component testing renders the UI as expected 1`] = `
<ul>
  <li
    className="session-participant"
  >
    <img
      alt="user img"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6>
        Axe Kubicka
      </h6>
      <div
        className="chat-message"
      >
        Hello world
      </div>
    </div>
  </li>
  <li
    className="session-participant"
  >
    <div
      className="participant-info"
    >
      <div
        className="chat-message"
      >
        Caty Flucker gave a thumbs up
      </div>
    </div>
  </li>
  <li
    className="session-participant"
  >
    <div
      className="participant-info"
    >
      <div
        className="chat-message"
      >
        Frank Runciman gave a thumbs down
      </div>
    </div>
  </li>
  <li
    className="session-participant"
  >
    <div
      className="participant-info"
    >
      <div
        className="chat-message"
      >
        Koren Templeton raises their hand
      </div>
    </div>
  </li>
  <li
    className="session-participant"
  >
    <div
      className="participant-info"
    >
      <div
        className="chat-message"
      >
        Koren Templeton claps
      </div>
    </div>
  </li>
  <li
    className="session-participant"
  >
    <div
      className="participant-info"
    >
      <div
        className="chat-message"
      >
        Koren Templeton joins
      </div>
    </div>
  </li>
  <li
    className="session-participant"
  >
    <div
      className="participant-info"
    >
      <div
        className="chat-message"
      >
        Ashla Attwool leave
      </div>
    </div>
  </li>
  <li
    className="session-participant"
  >
    <div
      className="participant-info"
    >
      <div
        className="chat-message"
      >
        Axe Kubicka joins the stage
      </div>
    </div>
  </li>
  <li
    className="session-participant"
  >
    <div
      className="participant-info"
    >
      <div
        className="chat-message"
      >
        Axe Kubicka leaves the stage
      </div>
    </div>
  </li>
</ul>
`;
